File: app/src/main/java/com/foxek/simpletimer/ui/workout/WorkoutContact.java
package com.foxek.simpletimer.ui.workout;

import com.foxek.simpletimer.data.model.Workout;
import com.foxek.simpletimer.ui.base.MvpInteractor;
import com.foxek.simpletimer.ui.base.MvpPresenter;
import com.foxek.simpletimer.ui.base.MvpView;

import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.disposables.Disposable;

public interface WorkoutContact {

    interface View extends MvpView {

        void startIntervalActivity(int position, String name);

        void setWorkoutList();

        void renderWorkoutList(List<Workout> workoutList);

        void showCreateDialog();
    }

    interface Presenter extends MvpPresenter<View, Interactor> {

        void saveButtonClicked(String workoutName);

        void createButtonClicked();

        void onListItemClicked(Workout workout);

    }

    interface Interactor extends MvpInteractor {

        Flowable<List<Workout>> fetchWorkoutList();

        Disposable createWorkout(String workoutName);
    }
}

File: app/src/main/java/com/foxek/simpletimer/ui/workout/WorkoutContact.kt
package com.foxek.simpletimer.ui.workout

import com.foxek.simpletimer.data.model.Workout
import com.foxek.simpletimer.ui.base.MvpPresenter
import com.foxek.simpletimer.ui.base.MvpView

import io.reactivex.Flowable
import io.reactivex.disposables.Disposable

interface WorkoutContact {

    interface View : MvpView {

        fun startIntervalActivity(position: Int, name: String)

        fun setWorkoutList()

        fun renderWorkoutList(workoutList: List<Workout>)

        fun showCreateDialog()
    }

    interface Presenter : MvpPresenter<View> {

        fun saveButtonClicked(name: String)

        fun createButtonClicked()

        fun onListItemClicked(item: Workout)

    }

    interface Interactor {

        fun fetchWorkoutList(): Flowable<List<Workout>>

        fun createWorkout(name: String): Disposable
    }
}

